modules/repository-s3/src/yamlRestTest/java/org/elasticsearch/repositories/s3/RepositoryS3ClientYamlTestSuiteIT.java

```diff
@@ -8,20 +8,158 @@
+            );
+            cluster.configFile("repository-s3/aws-web-identity-token-file", Resource.fromClasspath("aws-web-identity-token-file"))
+                .environment("AWS_WEB_IDENTITY_TOKEN_FILE", System.getProperty("awsWebIdentityTokenExternalLocation"))
+                // The AWS STS SDK requires the role and session names to be set. We can verify that they are sent to S3S in the
+                // S3HttpFixtureWithSTS fixture
+                .environment("AWS_ROLE_ARN", "arn:aws:iam::123456789012:role/FederatedWebIdentityRole")
+                .environment("AWS_ROLE_SESSION_NAME", "sts-fixture-test");
+        } else if (TEST_TASK.endsWith("yamlRestTestRegionalSTS")) {
+            cluster.configFile("repository-s3/aws-web-identity-token-file", Resource.fromClasspath("aws-web-identity-token-file"))
+                .environment("AWS_WEB_IDENTITY_TOKEN_FILE", System.getProperty("awsWebIdentityTokenExternalLocation"))
+                // The AWS STS SDK requires the role and session names to be set. We can verify that they are sent to S3S in the
+                // S3HttpFixtureWithSTS fixture
+                .environment("AWS_ROLE_ARN", "arn:aws:iam::123456789012:role/FederatedWebIdentityRole")
+                .environment("AWS_ROLE_SESSION_NAME", "sts-fixture-test")
+                .environment("AWS_STS_REGIONAL_ENDPOINTS", "regional")
+                .environment("AWS_REGION", "ap-southeast-2");
+        } else if (TEST_TASK.endsWith("yamlRestTest")) {
+            cluster.keystore("s3.client.integration_test_permanent.access_key", System.getProperty("s3PermanentAccessKey"))
+                .keystore("s3.client.integration_test_permanent.secret_key", System.getProperty("s3PermanentSecretKey"))
+                .keystore("s3.client.integration_test_temporary.access_key", System.getProperty("s3TemporaryAccessKey"))
+                .keystore("s3.client.integration_test_temporary.secret_key", System.getProperty("s3TemporarySecretKey"))
+                .keystore("s3.client.integration_test_temporary.session_token", s3TemporarySessionToken);
+            if (USE_FIXTURE) {
+                cluster.setting("s3.client.integration_test_permanent.endpoint", () -> s3Environment.getAddress(S3Fixture))
+                    .setting("s3.client.integration_test_temporary.endpoint", () -> s3Environment.getAddress(S3FixtureWithToken))
+                    .setting("s3.client.integration_test_ec2.endpoint", () -> s3Environment.getAddress(S3FixtureEc2))
+                    .systemProperty("com.amazonaws.sdk.ec2MetadataServiceEndpointOverride", () -> s3Environment.getAddress(S3FixtureEc2));
+            }
+        }
+
+        return cluster.build();
     }
 
+    public static TestRule ruleChain = RuleChain.outerRule(s3Environment).around(cluster);
+
+    @ClassRule
+    public static TestRule ruleChain2 = RuleChain.outerRule(minioFixtureTestContainer).around(ruleChain);
```

Comment:
No need to have multiple rule chains. You can call `around` multiple times. This can be simplified to:
```
RuleChain.outerRule(minioFixtureTestContainer).around(s3Environment).around(cluster);
```

Reply:
fixed